packages/beacon-node/src/chain/regen/regen.ts

```diff
@@ -242,18 +242,19 @@ async function processSlotsToNearestCheckpoint(
   const preSlot = preState.slot;
   const postSlot = slot;
   const preEpoch = computeEpochAtSlot(preSlot);
-  let postState = preState.clone();
+  let postState = preState;
   const {checkpointStateCache, emitter, metrics} = modules;
 
   for (
     let nextEpochSlot = computeStartSlotAtEpoch(preEpoch + 1);
     nextEpochSlot <= postSlot;
     nextEpochSlot += SLOTS_PER_EPOCH
   ) {
+    // processSlots calls .clone() before mutating
     postState = processSlots(postState, nextEpochSlot, metrics);
 
     // Cache state to preserve epoch transition work
-    const checkpointState = postState.clone();
+    const checkpointState = postState;
```

Comment:
@dapplion I think it's safer to do `postState.clone(false)`?

from your implementation

```typescript
  clone(dontTransferCache?: boolean): this {
    if (dontTransferCache) {
      return this.type.getViewDU(this.node) as this;
    } else {
      const cache = this.cache;
      this.clearCache();
      return this.type.getViewDU(this.node, cache) as this;
    }
  }
```

Reply:
Sorry I don't follow, what would it be safer to do a clone here?

Reply:
@tuyennhv The point of this PR is to think critically about when our state can actually be mutated. My opinion is that we clone is too many places that a state can't be mutated so cloning is an unnecessary expense that slows than Lodestar for no gain

Reply:
@dapplion in the past we used to have an issue of missing a `clone()` call, it really took time to debug in that case, @wemeetagain may experience it

so what I mean is instead of dropping the `clone()` call, can we do `clone(false)` with less performance impact?

if both `clone()` and `clone(false)` cause performance issue then we need to benchmark to improve performance?

Reply:
My confusion: I thought the cache is merely used for the mutation, it also contains the index of nodes so that we only have to traverse to the node once.

if we use `clone(false)`, the consumer does not have a chance to use the cached `nodes` of the tree so it's all bad going with either `clone() or clone(false)`

with ssz v2, we should not  ever have a state mutation unless in state transition, thanks @dapplion for your explanation 👍